primitives/core/src/crypto.rs

```diff
@@ -647,6 +647,28 @@ impl<'de> serde::Deserialize<'de> for AccountId32 {
 	}
 }
 
+#[cfg(feature = "std")]
+impl sp_std::str::FromStr for AccountId32 {
+	type Err = &'static str;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		if &s[..2] == "0x" && s.len() == 66 {
```

Comment:
This can panic if 2 is within one unicode character, better use `starts_with`.
You can also use `let s = s.trim_start_matches("0x");` to get rid of one branch.